src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -443,17 +451,18 @@ protected void InitializeReferences()
                 // If runmatch is null it means that an override of Scan didn't implement it correctly, so we will
                 // let this null ref since there are lots of ways where you can end up in a erroneous state.
                 Match match = runner.runmatch!;
-                if (match!.FoundMatch)
+                if (match.FoundMatch)
                 {
-                    if (quick)
+                    if (mode == RegexRunnerMode.ExistenceRequired)
                     {
-                        return null;
+                        return (true, 0, 0, 0);
```

Comment:
NIT: would it make more sense to have this return `-1` to indicate the info wasn't calculated? I suppose it is possible to ask for captures and get a 0-length match at index 0 and have the textpos to also be 0, so it would probably be helpful to differentiate from that.

Reply:
Ok